test/test_integration_single.rb

```diff
@@ -254,4 +254,30 @@ def test_puma_debug_loaded_exts
 
     cli_pumactl 'stop'
   end
+
+  def test_pre_existing_unix_after_idle_timeout
+    skip_unless :unix
+
+    File.open(@bind_path, mode: 'wb') { |f| f.puts 'pre existing' }
+
+    cli_server "-q test/rackup/hello.ru", unix: :unix, config: "idle_timeout 1"
+
+    sleep 1.15 # ensure the timeout isn't set until the first connection
+
+    sock = connection = connect(nil, unix: true)
+    read_body(connection)
+
+    sleep 1.15
+
+    assert sock.wait_readable(1), 'Unexpected timeout'
+    assert_raises Errno::ECONNREFUSED do
```

Comment:
Looks like JRuby raises `IOError` with message `Connection refused` (e.g. https://github.com/puma/puma/actions/runs/5847300372/job/15853513628#step:8:809)

Reply:
Cheers, I've handled that with 3e5d42ffdb1810c65a188000af7ddcfad9178e78.